src/Altinn.Platform/Altinn.Platform.Events/Events/Repository/PostgresRepository.cs

```diff
@@ -65,6 +68,84 @@ public async Task<string> Create(CloudEvent item)
             }
         }
 
+        /// <inheritdoc/>
+        public async Task<int> CreateEventsSubscription(EventsSubscription eventsSubscription)
+        {
+            try
+            {
+                await _conn.OpenAsync();
+
+                NpgsqlCommand pgcom = new NpgsqlCommand(insertSubscriptionSql, _conn);
+                pgcom.Parameters.AddWithValue("sourcefilter", eventsSubscription.SourceFilter);
+
+                if (eventsSubscription.SubjectFilter != null)
+                {
+                    pgcom.Parameters.AddWithValue("subjectfilter", eventsSubscription.SubjectFilter);
+                }
+                else
+                {
+                    pgcom.Parameters.AddWithValue("subjectfilter", DBNull.Value);
+                }
+
+                if (eventsSubscription.TypeFilter != null)
+                {
+                    pgcom.Parameters.AddWithValue("typefilter", eventsSubscription.TypeFilter);
+                }
+                else
+                {
+                    pgcom.Parameters.AddWithValue("typefilter", DBNull.Value);
+                }
+
+                int vid = 0;
+                pgcom.Parameters.AddWithValue("consumer", eventsSubscription.Consumer);
+                pgcom.Parameters.AddWithValue("endpointurl", eventsSubscription.EndPoint);
+                pgcom.Parameters.AddWithValue("createdby", eventsSubscription.CreatedBy);
+                pgcom.Parameters.AddWithValue("validated", false);
+                pgcom.Parameters.AddWithValue("v_id", vid);
+             
+                using (NpgsqlDataReader reader = pgcom.ExecuteReader())
+                {
+                    while (reader.Read())
+                    {
+                        eventsSubscription.Id = int.Parse(reader["v_id"].ToString());
+                    }
+                }
+
+                return eventsSubscription.Id;
+            }
+            catch (Exception e)
+            {
+                _logger.LogError("PostgresRepository // Create // Exception", e);
+                throw;
+            }
+            finally
+            {
+                await _conn.CloseAsync();
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task DeleteSubscription(int id)
+        {
+            try
+            {
+                await _conn.OpenAsync();
+                NpgsqlCommand pgcom = new NpgsqlCommand(deleteSubscription, _conn);
+                pgcom.Parameters.AddWithValue("_id", id);
+             
+                await pgcom.ExecuteNonQueryAsync();
+            }
+            catch (Exception e)
+            {
+                _logger.LogError("PostgresRepository // Create // Exception", e);
```

Comment:
update comment here so it is easier to identify which method produced the error.

Reply:
Fixed